packages/frontend/editor-ui/src/stores/focusPanel.store.ts

```diff
@@ -1,22 +1,45 @@
 import { STORES } from '@n8n/stores';
 import { defineStore } from 'pinia';
-import { ref } from 'vue';
+import { computed, ref } from 'vue';
+import get from 'lodash/get';
 
-import type { INodeProperties } from 'n8n-workflow';
+import { type NodeParameterValueType, type INode, type INodeProperties } from 'n8n-workflow';
+import { useWorkflowsStore } from './workflows.store';
 
 type FocusedNodeParameter = {
-	nodeName: string;
+	nodeId: string;
 	parameter: INodeProperties;
 	parameterPath: string;
-	value: string;
+};
+
+export type RichFocusedNodeParameter = FocusedNodeParameter & {
+	node: INode;
+	value: NodeParameterValueType;
 };
 
 export const useFocusPanelStore = defineStore(STORES.FOCUS_PANEL, () => {
+	const workflowsStore = useWorkflowsStore();
+
 	const focusPanelActive = ref(false);
-	const focusedNodeParameters = ref<FocusedNodeParameter[]>([]);
+	const _focusedNodeParameters = ref<FocusedNodeParameter[]>([]);
+
+	// An unenriched parameter indicates a missing nodeId
+	const focusedNodeParameters = computed<Array<RichFocusedNodeParameter | FocusedNodeParameter>>(
```

Comment:
I wonder if there is more efficient way to get parameter value here instead of iterating over parameters and getting full node from the store in every iteration.